charts/tekton-pipelines/templates/buildpacks/tasks/set-env-vars.yaml

```diff
@@ -0,0 +1,88 @@
+{{ if .Values.buildpacks.enabled }}
+
+# I dislike this hack
+# https://github.com/tektoncd/catalog/blob/main/task/buildpacks/0.3/buildpacks.yaml#L68-L113
+# but we should use it until this is resolved:
+# - https://github.com/buildpacks/tekton-integration/issues/33
+# - https://github.com/buildpacks/tekton-integration/issues/32
+
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: buildpack-set-env-vars
+  namespace: {{ .Release.Namespace }}
+spec:
+  description: >-
+    The task parses project.toml file and stores its environment variables
+    values to platform dir for further processing by buildpack task.
+
+  workspaces:
+    - name: source
+
+  resources:
+    inputs:
+      - name: app
+        type: git
+
+  params:
+    - name: application
+      type: string
+      description: name of the argocd application we're going to deploy/sync
+    
+    - name: platform_dir
+      type: string
+      description: The name of the platform directory.
+      
+  steps:
+    - name: clean
+      image: {{ .Values.images.bash | default "docker.io/library/bash:latest" }}
+      imagePullPolicy: {{ .Values.imagePullPolicy }}
+      script: |
+        #!/usr/bin/env bash
+        # clean platform env dir before new run, so when env var is removed from project.toml file 
+        # it wouldn't influence the build
+        set -e
+
+        ENV_PATH=$(workspaces.source.path)/$(params.platform_dir)/env
+        
+        if [ -d $ENV_PATH ]; then
```

Comment:
You used not actual version, please compare it with repo - https://github.com/saritasa-nest/saritasa-devops-tekton/blob/main/tekton-pipelines/base/tasks/buildpack-env-vars-task.yaml

Reply:
I improved your latest version. I don't see anything that I left behind
![image](https://user-images.githubusercontent.com/70822710/139955417-994aa1c1-d013-4938-80a3-0a9c4ef41565.png)

- workingDir: $(resources.inputs.kubernetes-repo.path)   allows to remove `cd` commands 
- `if [ ! `git status --porcelain` ]; then` were failing in alpine's if it returns non-empty list of files, I had to convert it to `if [ $(git status --porcelain | wc -l) -eq 0 ]; then`


If I missed something - pls let me know more details.

Reply:
I've synced from a PR branch instead.